__tests__/utils/cache.test.js

```diff
@@ -0,0 +1,31 @@
+import {act} from 'react-test-renderer';
+import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
+
+import {getSpecificCache, removeAllCache, saveToCache} from '../../src/utils/cache';
+
+describe('utils cache should run correctly', () => {
+  it('remove all cache shoule run correctly', () => {
+    const mockRemoveCache = jest.spyOn(mockAsyncStorage, 'clear');
+    act(() => {
+      removeAllCache();
+    });
+    expect(mockRemoveCache).toHaveBeenCalled();
+  });
+  it('saveToCache should run correctly', () => {
+    const callback = jest.fn().mockImplementation(() => undefined, 'cool');
+
+    const mockgetItem = jest
+      .spyOn(mockAsyncStorage, 'getItem')
+      .mockImplementation(('cache', callback));
+    act(() => {
+      saveToCache('cache', {name: 'agita'});
+    });
+    expect(mockgetItem).toHaveBeenCalled();
+  });
+  it('saveToCache should return json vale', async () => {
+    await saveToCache('data', {name: 'agita'});
+    await getSpecificCache('data', (data) => {
+      expect(data).toEqual({name: 'agita'});
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this test, you're saving data to the cache and then retrieving it. However, you're not mocking the `setItem` and `getItem` methods of `mockAsyncStorage`, so the test might fail if there's an issue with the AsyncStorage package itself rather than your code. You should mock these methods to isolate your code from external dependencies.
```diff
- await saveToCache('data', {name: 'agita'});
- await getSpecificCache('data', (data) => {
-   expect(data).toEqual({name: 'agita'});
- });
+ const mocksetItem = jest.spyOn(mockAsyncStorage, 'setItem').mockImplementation((key, value, callback) => callback(null));
+ const mockgetItem = jest.spyOn(mockAsyncStorage, 'getItem').mockImplementation((key, callback) => callback(null, JSON.stringify({name: 'agita'})));
+ 
+ await act(async () => {
+   await saveToCache('data', {name: 'agita'});
+   await getSpecificCache('data', (data) => {
+     expect(data).toEqual({name: 'agita'});
+   });
+ });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->